src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -244,35 +252,29 @@ ApplyLedgerChainWork::applyHistoryOfSingleLedger()
     }
 
     mApplyLedgerSuccess.Mark();
-    mLastApplied = hHeader;
     return true;
 }
 
 BasicWork::State
-ApplyLedgerChainWork::onRun()
+ApplyCheckpointWork::onRun()
 {
     try
     {
         if (!mFilesOpen)
         {
-            openCurrentInputFiles();
-        }
-
-        if (!applyHistoryOfSingleLedger())
-        {
-            mCurrSeq += mApp.getHistoryManager().getCheckpointFrequency();
-            mFilesOpen = false;
+            openInputFiles();
         }
 
-        mApp.getCatchupManager().logAndUpdateCatchupStatus(true);
+        auto result = applyHistoryOfSingleLedger();
+        auto const& lm = mApp.getLedgerManager();
+        auto done = lm.getLastClosedLedgerNum() == mCheckpointRange.mLast;
 
-        auto& lm = mApp.getLedgerManager();
-        auto const& lclHeader = lm.getLastClosedLedgerHeader();
-        if (lclHeader.header.ledgerSeq == mRange.mLast)
+        if (result)
         {
-            return State::WORK_SUCCESS;
+            mApp.getCatchupManager().logAndUpdateCatchupStatus(true);
```

Comment:
I'd call `logAndUpdateCatchupStatus` regardless (before the if) - it seems arbitrary to only call it if result is true.

Reply:
I would also move the `done` case to its own it (probably done first) as to make the code easier to read (else will be deciding between running or failure)

Reply:
done.